osu.Game/Skinning/Components/IconElement.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Newtonsoft.Json.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Localisation;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Localisation.SkinComponents;
+using osu.Game.Overlays.Settings;
+using osuTK;
+
+namespace osu.Game.Skinning.Components
+{
+    public partial class IconElement : Container, ISerialisableDrawable
+    {
+        private static readonly Type[] all_icon_types = new[]
+        {
+            typeof(FontAwesome.Regular),
+            typeof(FontAwesome.Solid),
+            typeof(FontAwesome.Brands),
+            typeof(OsuIcon),
+        };
+
+        private static readonly IconModel[] all_icons = all_icon_types.SelectMany(t => t.GetProperties(BindingFlags.Public | BindingFlags.Static))
+                                                                      .Select(p => new IconModel(p.Name, (IconUsage)p.GetValue(null)!))
+                                                                      .ToArray();
+
+        private SpriteIcon spriteIcon = null!;
+
+        public bool UsesFixedAnchor { get; set; }
+
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.Icon), SettingControlType = typeof(SettingsIconDropdown))]
+        public BindableIcon Icon { get; } = new BindableIcon(all_icons.First());
+
+        [SettingSource(typeof(SkinnableComponentStrings), nameof(SkinnableComponentStrings.IconColour), nameof(SkinnableComponentStrings.TextColourDescription))]
+        public BindableColour4 IconColour { get; } = new BindableColour4(Colour4.White);
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            Child = spriteIcon = new SpriteIcon
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Size = new Vector2(40),
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Icon.BindValueChanged(e => spriteIcon.Icon = e.NewValue.Icon, true);
+            IconColour.BindValueChanged(e => spriteIcon.Colour = e.NewValue, true);
+        }
+
+        public record IconModel(string Name, IconUsage Icon);
```

Comment:
I don't like this at all. Have you checked how this gets serialised out? It's probably going to allow people to do stupid unexpected shit.

You're also making `IconUsage` a *serliased type* for the first time. I'd be against this.